### PR TITLE
feat: Auth middleware

### LINKED ISSUE #12: [Blockchain] Authentication middleware for endpoints

### PR DESCRIPTION
**Pull request by Jatewo**
## Description
This pull request introduces several significant changes across multiple files, focusing on adding JSDoc comments for better documentation, setting up a new GitHub workflow for Hardhat tests, and enhancing authentication middleware. Additionally, there are updates to dependencies and scripts in `package.json`.

### Documentation Improvements:
* Added JSDoc comments to various functions in `controllers/candidate.js`, `controllers/election.js`, `controllers/party.js`, `controllers/vote.js`, `utils/convertBigIntToString.js`, `utils/startContract.js`, and `utils/stopContract.js` to improve code documentation. [[1]](diffhunk://#diff-fa19ce6ca5b95b57701840233055851d9b6ed0a861a6437e9e26afda0121253bR5-R11) [[2]](diffhunk://#diff-fa19ce6ca5b95b57701840233055851d9b6ed0a861a6437e9e26afda0121253bR27-R33) [[3]](diffhunk://#diff-bd3d0088bdc5e18fc4a47740bbac6fbdc2aae27771962007725dade8c20ac239R7-R13) [[4]](diffhunk://#diff-bd3d0088bdc5e18fc4a47740bbac6fbdc2aae27771962007725dade8c20ac239R50-R56) [[5]](diffhunk://#diff-6a2a983706566f7a13f55bcad93859a87f879275377cfb99e56f1096aa1d9d8dR5-R11) [[6]](diffhunk://#diff-6a2a983706566f7a13f55bcad93859a87f879275377cfb99e56f1096aa1d9d8dR27-R32) [[7]](diffhunk://#diff-23dbd3aeaf878b223d47ce6b8b35e4fe3d064c0185095e1be2ea8cb6b4cd4f9bR6-R12) [[8]](diffhunk://#diff-72090209f0e87cb8aca2562f55f6a06b5398af35e0dc8f14dd470d6a5e557997R1-R5) [[9]](diffhunk://#diff-1ad4da62101b3bb21fa70d12ceab20495fa1e39722e6d91f50f737db56bc11f3R6-R8) [[10]](diffhunk://#diff-7468699f5eb85a5edd62e04517916cbe7b0d09678dd8877d828236b0b35002e6R4-R6)

### Workflow Enhancements:
* Created a new GitHub Actions workflow file `.github/workflows/hardhat-test.yml` to automate running Hardhat tests on pull requests.
* Updated `.github/workflows/jest.yml` to include a step for compiling Hardhat before running Jest tests.

### Authentication Middleware:
* Added a new middleware `auth.js` for JWT-based authentication and corresponding tests in `auth.test.js`. [[1]](diffhunk://#diff-e135f48acbdd2de273a36dcabccdbe0da897b962814b48b60aa1ea8af86034d8R1-R27) [[2]](diffhunk://#diff-ec7c9e8435be7c486eac7ee791c1b245ad9260d7b2a2b8e36805f2cf76391a51R1-R59)

### Configuration and Dependencies:
* Updated `eslint.config.js` to include `eslint-plugin-jsdoc` for enforcing JSDoc rules. [[1]](diffhunk://#diff-a32a0887ed9d1d707bbb3b845b7df7fd40e673c47e7b60a3ebd896b68d3b8839R4-R9) [[2]](diffhunk://#diff-a32a0887ed9d1d707bbb3b845b7df7fd40e673c47e7b60a3ebd896b68d3b8839R20-R21)
* Modified `package.json` to add `dotenv` and `jsonwebtoken` dependencies, and added a new script for running Hardhat tests. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L8-R9) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L25-R35)

These changes collectively enhance code quality, testing automation, and security through improved documentation, continuous integration, and authentication mechanisms.

### CODE REVIEW

**Review by MortenMunk — DISMISSED** [review]
lgtm

File: controllers/candidate.js [hunk 1]
```diff
@@ -17,6 +24,13 @@ async function getCandidates(req, res, next) {
     }
 }
 
+/**
+ * Publish a candidate to the blockchain
+ * @param {String} objectId mongoose object id of the candidate
+ * @param {String} name The name of the candidate
+ * @param {String} party mongoose object id of the candidate's party
```
Comment (Fantaskink) [inline controllers/candidate.js]: Not party mongoose object id, it's the party name.

File: .github/workflows/jest.yml [hunk 2]
```diff
@@ -20,5 +20,8 @@ jobs:
       - name: Install dependencies
         run: npm i
 
+      - name: Compile Hardhat
```
Comment (Fantaskink) [inline .github/workflows/jest.yml]: Compile step is already in npm run test.

**Review by Fantaskink — CHANGES_REQUESTED** [review]

**Review by MortenMunk — DISMISSED** [review]

**Review by Fantaskink — APPROVED** [review]
lgtm

**Pull request merged**